packages/desktop-client/src/components/budget/rollover/CoverMenu.tsx

```diff
@@ -1,20 +1,43 @@
 import React, { useState } from 'react';
 
+import { type CategoryGroupEntity } from 'loot-core/types/models/category-group';
+
 import { useCategories } from '../../../hooks/useCategories';
 import { CategoryAutocomplete } from '../../autocomplete/CategoryAutocomplete';
 import { Button } from '../../common/Button';
 import { InitialFocus } from '../../common/InitialFocus';
 import { View } from '../../common/View';
 import { addToBeBudgetedGroup } from '../util';
 
+function removeSelectedCategory(
+  categoryGroups: CategoryGroupEntity[],
+  selectedCategoryId?: string,
+) {
+  if (!selectedCategoryId) return categoryGroups;
+
+  const newCategoryGroups: CategoryGroupEntity[] = JSON.parse(
```

Comment:
Question: Why do we need to parse/stringify?

Reply:
You're right, I was hitting issues with it changing the object in the higher scope as it was passed by ref. Reworked to use with `.map` now.